components/authz-service/server/v2/project_update_manager.go

```diff
@@ -523,3 +524,233 @@ func getEstimatedTimeCompleteInSec(event *automate_event.EventMsg) (time.Time, e
+	if err == cereal.ErrWorkflowInstanceNotFound {
+		return nil
+	}
+	return err
+}
+
+func (m *CerealProjectUpdateManager) Start() error {
+	params := map[string]interface{}{}
+
+	for _, svc := range m.domainServices {
+		params[svc] = project_update_tags.DomainProjectUpdateWorkflowParameters{
+			ProjectUpdateID: "project-update-id",
+		}
+	}
+	return multiworkflow.EnqueueWorkflow(context.Background(), m.manager, m.workflowName, m.instanceName, params)
+}
+
+func (m *CerealProjectUpdateManager) Failed() bool {
+	instance, err := multiworkflow.GetWorkflowInstance(context.Background(), m.manager, m.workflowName, m.instanceName)
+	if err != nil {
+		if err == cereal.ErrWorkflowInstanceNotFound {
+			return false
+		}
+		logrus.WithError(err).Error("failed to get workflow instance")
+		return true
+	}
+
+	if instance.Err() != nil {
+		return true
+	}
+
+	var payload *multiworkflow.MultiWorkflowPayload
+	if instance.IsRunning() {
+		var err error
+		payload, err = instance.GetPayload()
+		if err != nil {
+			logrus.WithError(err).Error("failed to get workflow instance payload")
+			return true
+		}
+	} else {
+		var err error
+		payload, err = instance.GetResult()
+		if err != nil {
+			logrus.WithError(err).Error("failed to get workflow instance payload")
+			return true
+		}
+	}
+
+	// OnStart has not run yet
+	if payload == nil {
+		return false
+	}
+
+	for _, d := range m.domainServices {
+		subWorkflow, err := instance.GetSubWorkflow(d)
+		if err != nil {
+			logrus.Warnf("subworkflow %q not found", d)
+			continue
+		}
+
+		if subWorkflow.Err() != nil {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (m *CerealProjectUpdateManager) FailureMessage() string {
+	instance, err := multiworkflow.GetWorkflowInstance(context.Background(), m.manager, m.workflowName, m.instanceName)
+	if err != nil {
+		if err == cereal.ErrWorkflowInstanceNotFound {
+			return ""
+		}
+		logrus.WithError(err).Error("failed to get workflow instance")
+		return err.Error()
+	}
+
+	if instance.Err() != nil {
+		return instance.Err().Error()
+	}
+
+	var payload *multiworkflow.MultiWorkflowPayload
+	if instance.IsRunning() {
+		var err error
+		payload, err = instance.GetPayload()
+		if err != nil {
+			logrus.WithError(err).Error("failed to get workflow instance payload")
+			return err.Error()
+		}
+	} else {
+		var err error
+		payload, err = instance.GetResult()
+		if err != nil {
+			logrus.WithError(err).Error("failed to get workflow instance payload")
+			return err.Error()
+		}
+	}
+
+	// OnStart has not run yet
+	if payload == nil {
+		return ""
+	}
+
+	errMsg := ""
+	for _, d := range m.domainServices {
+		subWorkflow, err := instance.GetSubWorkflow(d)
+		if err != nil {
+			logrus.Warnf("subworkflow %q not found", d)
+			continue
+		}
+
+		if subWorkflow.Err() != nil {
+			errMsg = fmt.Sprintf("%s; %s: %s", errMsg, d, subWorkflow.Err().Error())
+		}
+	}
+
+	return errMsg
+}
+
+func (m *CerealProjectUpdateManager) PercentageComplete() float64 {
```

Comment:
What we were doing before for PercentageComplete was using the domain service with the longest estimated time to complete's PercentageComplete. What might be easier would be to use the domain service with the lowest PercentageComplete. If there were three domain services with 10%, 35%, and 95% all running at the same time, we would return the entire process is 10% complete. Here we are assuming that the 35% and 95% tasks will complete before the 10% one. If the task does not complete before the others then their percentage will become the new PercentageComplete. 
I could not think of a way of adding the domain percentage completes together because we do not know how much work each of them has to do. The average does not really make sense to me because some of the tasks are going to be completed a lot faster than the others. For example, say we have three domain tasks that have been running for a minute. Two of the tasks are complete but the other one takes an hour. I would think the percentage would be 1/60 = 1.6% complete, but the below calculation would give 1/3+1/3+1/60/3 = 67.2%.

Reply:
if percentage should be tied to time, then # complete tasks doesn't make sense. I'll change it back to how it was before